crypto/cache.go

```diff
@@ -73,117 +63,64 @@ func (cache *cache) evict() {
 	if len(cache.entries) < cache.capacity {
 		return
 	}
-	key := cache.accessOrder.Remove(cache.accessOrder.Back()).(key)
+	key := cache.accessOrder.Remove(cache.accessOrder.Back()).(string)
 	delete(cache.entries, key)
 }
 
-// Sign signs a hash.
-func (cache *cache) Sign(hash consensus.Hash) (sig consensus.Signature, err error) {
-	sig, err = cache.impl.Sign(hash)
+// Sign signs a message.
+func (cache *cache) Sign(message []byte) (sig consensus.QuorumSignature, err error) {
+	sig, err = cache.impl.Sign(message)
 	if err != nil {
 		return nil, err
 	}
-	key := sha256.Sum256(append(hash[:], sig.ToBytes()...))
-	cache.insert(key, false)
+	var key strings.Builder
+	hash := sha256.Sum256(message)
+	_, _ = key.Write(hash[:])
```

Comment:
Suggest to add a comment to explain why we ignore checking these errors...

Reply:
It is because writing to a string builder never returns an error. It is just immplementing the io.Writer interface.

Reply:
Reasonable. Probably fine to not add a comment then.